src/Core/Mesh/TriangleMesh.inl

```diff
@@ -36,55 +32,132 @@ inline TriangleMesh& TriangleMesh::operator=( TriangleMesh&& other ) {
     m_vertexAttribs = std::move( other.m_vertexAttribs );
     m_verticesHandle = std::move( other.m_verticesHandle );
     m_normalsHandle = std::move( other.m_normalsHandle );
+    return *this;
 }
 
-inline void TriangleMesh::clear() {
-    m_vertexAttribs.clear();
-    vertices().clear();
-    normals().clear();
-    m_triangles.clear();
-
-    initDefaultAttribs();
-}
+inline bool TriangleMesh::append( const TriangleMesh& other ) {
+    // check same attributes through names
+    bool sameAttrib = true;
+    const auto& m_attr = m_vertexAttribs.m_attribsIndex;
+    const auto& o_attr = other.m_vertexAttribs.m_attribsIndex;
+    // one way
+    for ( const auto& attr : m_attr )
+    {
+        if ( o_attr.find( attr.first ) == o_attr.end() )
+        {
+            sameAttrib = false;
+            break;
+        }
+    }
+    if ( !sameAttrib )
+        return false;
+    // the other way
+    for ( const auto& attr : o_attr )
+    {
+        if ( m_attr.find( attr.first ) == m_attr.end() )
+        {
+            sameAttrib = false;
+            break;
+        }
+    }
+    if ( !sameAttrib )
+        return false;

```

Comment:
The computation of `sameAttrib` is generic and must be implemented in `AttribManager`, not in `TriangleMesh`.